routers/api/v1/users.go

```diff
@@ -347,3 +347,124 @@ func (r *apiV1Router) VerifyEmail(ctx *gin.Context) {
 		Status: http.StatusOK,
 	})
 }
+
+// GET: /api/v1/users/password/reset?email={email}
+// Request:  email string
+// Response: status int
+//           error string
+func (r *apiV1Router) GetPasswordResetEmail(ctx *gin.Context) {
+	email := ctx.Query("email")
+	if len(email) == 0 {
+		r.logger.Warn("email not specified")
+		models.SendAPIError(ctx, http.StatusBadRequest, "email must be specified")
+		return
+	}
+
+	user, err := r.userService.GetUserWithEmail(ctx, email)
+	if err != nil {
+		if err == services.ErrNotFound {
+			r.logger.Warn("user with email doesn't exist", zap.String("email", email))
+			// don't want to give the user a tool to figure which email addresses are registered
+			// as this would be a security issue
+			ctx.JSON(http.StatusOK, models.Response{
+				Status: http.StatusOK,
+			})
+			return
+		}
+		r.logger.Error("could not fetch user", zap.String("email", email), zap.Error(err))
+		models.SendAPIError(ctx, http.StatusInternalServerError, "something went wrong")
+		return
+	}
+
+	emailToken, err := auth.NewJWT(*user, time.Now().Unix(), 100, auth.Email, []byte(r.env.Get(environment.JWTSecret)))
```

Comment:
What is the duration for which this token is valid?

Reply:
This is currently no implemented but I've set it up for the future such that when we do implement token expiration, it's as easy as just modifying the methods that validate and create the token

Reply:
Okay, I think we it should be a priority since currently the token is valid forever so a user can change their password multiple times with the same reset email.